### DISCUSSION TITLE
How can I prevent divergence with adaptive timesteps?

### DISCUSSION THREAD

**Discussion by ke7kto** [idaholab/moose#17508] [Q&A General] [answered]
I'm working on adding an adaptive timestep to my simulation, but I'm getting convergence issues. I then tried applying adaptive timesteppers to example 8, which causes the executable to error out because of divergence at any step size. 

I've tried AB2PredictorCorrector and DT2 for the adaptive timesteps, and both end with errors. Is there some sort of a tutorial on appropriate settings for these timesteppers, and what solvers/preconditioners work well with them? Or on common pitfalls that should be avoided?

[ex08_adaptive.i.txt](https://github.com/idaholab/moose/files/6259353/ex08_adaptive.i.txt)
[ex08ab2.i.txt](https://github.com/idaholab/moose/files/6259354/ex08ab2.i.txt)

**Comment by ke7kto** [comment, marked as answer]
Seems like the parameter `nl_abs_tol` was helpful to put in the `Executioner` block, as mentioned [here](https://mooseframework.inl.gov/application_usage/failed_solves.html). With this set to an arbitrary value, I do have convergence.

**Reply by WilkAndy** [reply]
Hopefully not too arbitrary, heheheh
